iwyu_output.cc

```diff
@@ -1774,29 +1774,45 @@ OutputLine PrintableIncludeOrForwardDeclareLine(
     GetSymbolsSortedByFrequency(line.symbol_counts()));
 }
 
-typedef pair<int, string> LineSortKey;
+enum class LineSortOrdinal {
+  kPrecompiledHeader,
+  kAssociatedHeader,
+  kAssociatedInlineDefinitions,
+  kCHeader,
+  kCppHeader,
+  kOtherHeader,
+  kForwardDeclaration
+};
+
+using LineSortKey = pair<LineSortOrdinal, string>;
+
+LineSortOrdinal GetLineSortOrdinal(const OneIncludeOrForwardDeclareLine& line,
+                                   const set<string>& associated_quoted_includes,
+                                   const IwyuFileInfo* file_info) {
+  if (!line.IsIncludeLine())
+    return LineSortOrdinal::kForwardDeclaration;
+  if (file_info && file_info->is_pch_in_code())
+    return LineSortOrdinal::kPrecompiledHeader;
+
+  const auto& quotedInclude = line.quoted_include();
```

Comment:
No `auto` if the type is not visible on the rhs of the assignment, please. Except in emergency cases like map iteration :)

Reply:
And I'd prefer the concrete type to be `std::string` (sans reference) -- I find lifetime extension to be a wart on the language (among many)

Reply:
Sure thing.